core/consensus/babe/primitives/src/digest.rs

```diff
@@ -102,41 +121,40 @@ impl Decode for BabePreDigest {
 }
 
 /// A digest item which is usable with BABE consensus.
-#[cfg(feature = "std")]
+// #[cfg(feature = "std")]
 pub trait CompatibleDigestItem: Sized {
 	/// Construct a digest item which contains a BABE pre-digest.
-	fn babe_pre_digest(seal: BabePreDigest) -> Self;
+	fn babe_pre_digest<D: Codec>(seal: D) -> Self;
```

Comment:
Why is this change needed? The runtime should be using `RawBabePreDigest` anyway. This function signature doesn't make a lot of sense, it looks like you can give it anything and it will make a BABE seal which is not the case at all, since you need to give it some specific struct to make a proper BABE seal.

Reply:
Yeah, it should be a better way to do it. But it also using a non raw pre digest.